### DISCUSSION TITLE
[F] As a developer it would be good if each row could be streamed when reading an XLSX file so that I could reduce the memory burden on my server when reading large files

### DISCUSSION THREAD

**Discussion by TankDoor** [exceljs/exceljs#2458] [Ideas & Proposals]
## 🚀 Feature Proposal

Stream each row of an XLSX file independently
row could be streamed when reading an XLSX file so that I could reduce the memory burden on my server when reading large files

## Motivation

 Reduce the memory burden on my server when reading large files

## Example

Reading large files on the server is utilizing large amounts of memory as all the customer's data is in the first sheet. As a result I am effectively without the benefit of streaming as the current process streams one sheet at a time

**Comment by prandell** [comment]
Second this. It does not appear there is any way to actually stream each row at a time from a workbook using the Workbook reader. Using stream handlers I still reach Heap Memory Exhaustion.
